src/roles/check_hostname/tasks/main.yaml

```diff
@@ -0,0 +1,25 @@
+---
+- name: Assert FQDN is not 'localhost' or 'localhost.localdomain'
+  ansible.builtin.assert:
+    that:
+      - ansible_fqdn != "localhost"
+      - ansible_fqdn != "localhost.localdomain"
+    fail_msg: "FQDN {{ ansible_fqdn }} is invalid: it must not be 'localhost' or 'localhost.localdomain'"
+
+- name: Assert FQDN contains a dot
+  ansible.builtin.assert:
+    that:
+      - ansible_fqdn.count('.') >= 1
+    fail_msg: "FQDN {{ ansible_fqdn }} must contain at least one dot."
+
+- name: Assert FQDN has no underscores
+  ansible.builtin.assert:
+    that:
+      - "'_' not in ansible_fqdn"
+    fail_msg: "FQDN {{ ansible_fqdn }} must not contain underscores"
```

Comment:
Out of curiosity, why did you use `count >= 1` (and not `in`) for the dot, but `not in` (and not `count == 0`) for the underscore?
Obviously either works, but I am curious :)